AEPMessaging/Sources/Messaging+EdgeEvents.swift

```diff
@@ -0,0 +1,220 @@
+                guard let messageProfile = MessagingConstants.AdobeTrackingKeys.MESSAGE_PROFILE_JSON.toJsonDictionary() else {
+                    Log.warning(label: MessagingConstants.LOG_TAG,
+                                "Failed to update xdmMap with adobe/cjm informations:" +
+                                    "converting message profile string to dictionary failed in the event '\(event.id.uuidString)'.")
+                    return xdmDictResult
+                }
+                // Merging the dictionary
+                cjmDict.mergeXdm(rhs: messageProfile)
+                experienceDict[MessagingConstants.AdobeTrackingKeys.CUSTOMER_JOURNEY_MANAGEMENT] = cjmDict
+                xdmDictResult[MessagingConstants.AdobeTrackingKeys.EXPERIENCE] = experienceDict
+            }
+        } else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Failed to send adobe/cjm information data with the tracking," +
+                            "\(MessagingConstants.AdobeTrackingKeys.EXPERIENCE) is missing in the event '\(event.id.uuidString)'.")
+        }
+        return xdmDictResult
+    }
+
+    /// Adding application data based on the application opened or not
+    /// - Parameters:
+    ///   - applicationOpened: `Bool` stating whether the application is opened or not
+    ///   - xdmData: `[AnyHashable: Any]` xdm data in which application data needs to be added
+    /// - Returns: `[String: Any]` which contains the application data
+    private func addApplicationData(applicationOpened: Bool, xdmData: [String: Any]) -> [String: Any] {
+        var xdmDataResult = xdmData
+        xdmDataResult[MessagingConstants.AdobeTrackingKeys.APPLICATION] =
+            [MessagingConstants.AdobeTrackingKeys.LAUNCHES:
+                [MessagingConstants.AdobeTrackingKeys.LAUNCHES_VALUE: applicationOpened ? 1 : 0]]
+        return xdmDataResult
+    }
+
+    /// Creates the xdm schema from event data
+    /// - Parameters:
+    ///   - event: `Event` with push notification tracking information
+    ///   - config: `[AnyHashable: Any]` with configuration informations
+    /// - Returns: `[String: Any]?` which contains the xdm data
+    private func getXdmData(event: Event, config: [AnyHashable: Any]) -> [String: Any]? {
+        guard let eventType = event.eventType else {
+            Log.warning(label: MessagingConstants.LOG_TAG, "Updating xdm data for tracking failed, eventType is invalid or nil in the event '\(event.id.uuidString)'.")
+            return nil
+        }
+        let messageId = event.messagingId
+        let actionId = event.actionId
+
+        if eventType.isEmpty == true || messageId == nil || messageId?.isEmpty == true {
+            Log.trace(label: MessagingConstants.LOG_TAG, "Updating xdm data for tracking failed, EventType or MessageId received in the event '\(event.id.uuidString)' is nil.")
+            return nil
+        }
+
```

Comment:
the above two if statements can be simplified

        guard let eventType = event.eventType, !eventType.isEmpty else {
            // log that eventType is invalid
            return nil
        }
        
        guard let messageId = event.messagingId, !messageId.isEmpty {
            // log that messageId is invalid
            return nil
        }